src/sagemaker/session.py

```diff
@@ -4347,11 +4347,39 @@ def submit(request):
             if model_package_group_name is not None and not model_package_group_name.startswith(
                 "arn:"
             ):
-                _create_resource(
-                    lambda: self.sagemaker_client.create_model_package_group(
-                        ModelPackageGroupName=request["ModelPackageGroupName"]
+                model_package_groups = []
+                model_package_groups_response = self.sagemaker_client.list_model_package_groups(
+                    NameContains=request["ModelPackageGroupName"],
+                )
+                model_package_groups = (
+                    model_package_groups
+                    + model_package_groups_response["ModelPackageGroupSummaryList"]
+                )
+                next_token = model_package_groups_response.get("NextToken")
+
+                while next_token is not None and next_token != "":
+                    model_package_groups_response = self.sagemaker_client.list_model_package_groups(
+                        NameContains=request["ModelPackageGroupName"], NextToken=next_token
+                    )
+                    model_package_groups = (
+                        model_package_groups
+                        + model_package_groups_response["ModelPackageGroupSummaryList"]
+                    )
+                    next_token = model_package_groups_response.get("NextToken")
+
+                filtered_model_package_group = list(
+                    filter(
+                        lambda mpg: mpg.get("ModelPackageGroupName")
+                        == request["ModelPackageGroupName"],
+                        model_package_groups,
```

Comment:
This logic will cause latency issuewhere if account has similar naming group. Cant we use search which has equals operation

Reply:
added search to query the model package group
and added list as a fallback if search fails